[WIP]Голографические карты: Эпизод 2 - Новая Надежда

Filatelele opened the pull request:
Еще более многострадальный порт голографических карт с /VG/

Воксы, нюкеры, обр и дедсад получили свои голографические чипы, которые встраиваются в шлем. Достать их можно с помощью отвертки. Как и раньше - отображают своих тиммейтов на карте. 

<details> 
  <summary>*Клик* </summary>
![](https://user-images.githubusercontent.com/23741266/35742698-5b722ce0-084c-11e8-8ae9-b7f5e1b9ef7e.png)

![](https://user-images.githubusercontent.com/23741266/35742486-a29d83f4-084b-11e8-8ed8-cf578f14fe6f.png)

![](https://user-images.githubusercontent.com/23741266/35742467-98fc30ca-084b-11e8-90f7-aa33bfc66c17.png)

![](https://user-images.githubusercontent.com/23741266/35742457-8fc78194-084b-11e8-917c-419c71b4c9f4.png)
</details>

:cl:
 - rscadd[link]: Добавлены голографические карты.

Kurshan commented on code/modules/clothing/clothing.dm:
```diff
@@ -306,6 +306,46 @@ BLIND     // can't see anything
+		I.loc = src
+		holochip = I
+		holochip.holder = src
+		playsound(user.loc, 'sound/items/Screwdriver.ogg', 100, 1)
```
Пускай лучше источником будет user. (даже не спрашивай зачем, это давнее извращение и внутренняя надежда)

Kurshan commented on code/modules/clothing/clothing.dm:
```diff
@@ -306,6 +306,46 @@ BLIND     // can't see anything
+		holochip.holder = src
+		playsound(user.loc, 'sound/items/Screwdriver.ogg', 100, 1)
+		to_chat(user, "<span class='notice'>[user] modifies the [src] with the [holochip]</span>")
+	if(istype(I, /obj/item/weapon/screwdriver))
```
Else тогда уж

Kurshan commented on code/modules/clothing/clothing.dm:
```diff
@@ -306,6 +306,46 @@ BLIND     // can't see anything
+	var/obj/item/holochip/holochip = null
+
+/obj/item/clothing/head/helmet/space/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/holochip) && holochip == null)
```
`holochip == null`
Можно просто знак отрицания поставить жеж

Kurshan replied:
А так же как по мне, стоит проверку на отсутствие holochip пониже вставить, чтобы юзеру высвечивалась мессага, почему процедура вставки голочипа не прошла

Kurshan commented on code/modules/clothing/clothing.dm:
```diff
@@ -306,6 +306,46 @@ BLIND     // can't see anything
+	if(istype(I, /obj/item/weapon/screwdriver))
+		holochip.deactivate_holomap()
+		holochip.holder = null
+		holochip.forceMove(get_turf(src))
```
Или put_in_hands?

Kurshan commented on code/modules/clothing/spacesuits/alien.dm:
```diff
@@ -109,6 +109,11 @@
 	desc = "Hey, wasn't this a prop in \'The Abyss\'?"
 
+/obj/item/clothing/head/helmet/space/vox/pressure/atom_init()
+	.=..()
```
Пробелов не завезли

Kurshan commented on code/modules/clothing/spacesuits/alien.dm:
```diff
@@ -133,9 +143,13 @@
 	armor = list(melee = 60, bullet = 30, laser = 30, energy = 15, bomb = 30, bio = 30, rad = 30)
 
+/obj/item/clothing/head/helmet/space/vox/stealth/atom_init()
+	.=..()
```
И тут тоже

Kurshan commented on code/modules/clothing/spacesuits/alien.dm:
```diff
@@ -221,6 +235,11 @@
 	desc = "An alien helmet with enormous goggled lenses."
 
+/obj/item/clothing/head/helmet/space/vox/medic/atom_init()
+	.=..()
```
И тут

Kurshan commented on code/modules/clothing/spacesuits/ert.dm:
```diff
@@ -7,6 +7,11 @@
 	var/obj/machinery/camera/camera
 
+/obj/item/clothing/head/helmet/space/rig/ert/atom_init()
+	.=..()
```
И даже тут

Kurshan commented on code/modules/clothing/spacesuits/ert.dm:
```diff
@@ -44,6 +49,9 @@
 	armor = list(melee = 60, bullet = 65, laser = 55, energy = 45, bomb = 50, bio = 100, rad = 60)
 
+/obj/item/clothing/head/helmet/space/rig/ert/commander/atom_init()
+	.=..()
```
Зачем?

Kurshan commented on code/modules/holomaps/holomap_chip.dm:
```diff
@@ -0,0 +1,187 @@
+	desc = "A small holomap module, attached to helmets."
+	icon = 'icons/holomaps/holochips.dmi'
+	icon_state = "holochip"
+	var/role_filter = null
```
Хоть бы какие комментарии внёс, ну в самом деле же.

Kurshan commented on code/modules/holomaps/holomap_chip.dm:
```diff
@@ -0,0 +1,187 @@
+		default_holomap = image(generate_holo_map())
+	holomap_base = default_holomap
+	holochips += src
+	if(!default_holomap)
```
А как так может получиться, что это же условие может не сработать на 43 строчке?

Filatelele replied:
Так могло случиться, но я это пофиксил, а потом забыл убрать проверку.

Kurshan commented on code/modules/holomaps/holomap_lists.dm:
```diff
@@ -0,0 +1 @@
+var/global/list/holochips = list()
```
Отдельный файл для одной строчки?

Kurshan commented on code/modules/holomaps/holomap_chip.dm:
```diff
@@ -0,0 +1,187 @@
+#undef HOLOMAP_WALKABLE_TILE
+#undef HOLOMAP_CONCRETE_TILE
+
+//��������������� ���� ��� ������ �����. � ����� ����� ����� ��� ��� ����� ������� ������, �� ���-�� �� �����
```
ЗИС РАШН ЛЕТТЕРС ИС АНРИДБЛ ВИЗ АУР ТЕКСТ ЕНКОДИНГ

Kurshan commented on code/modules/holomaps/holomap_chip.dm:
```diff
@@ -0,0 +1,187 @@
+
+/obj/item/holochip/proc/generate_holo_map()
+	var/icon/holomap = icon('icons/holomaps/canvas.dmi', "blank")
+	for(var/i = 1 to ((2 * world.view + 1) * 32))
```
Кажется был макрос, который относительно быстро собирал лист всех тайлов.

Filatelele replied:
Я нашел только прок spiral_range(), но что-то по его коду я сомневаюсь, что он соберет лист всех тайлов быстрее

Kurshan replied:
Не то

LudwigVonChesterfield replied:
RANGE_TURFS

Kurshan reviewed (changes requested):
Не думаю, что это все моменты, надо будет на более свежую голову мне ещё раз посмотреть. А пока вопрос, чё профайлер говорит про нагрузку всех этих методов?

Filatelele committed:
Holomaps

Filatelele committed:
Merge branch 'master' into Holomaps

Filatelele committed:
Bugfixes

list/holochips был перенесен в  holomap_chip.dm
В атом_ините появились точки
Звук проигрывается из user, а не из user.loc

Filatelele commented:
Насчет профайлера. Посмотрел, никаких аномальных йоба-нагрузок не обнаружил. Но я себе не очень доверяю, так что лучше, чтобы кто-нибудь еще посмотрел.

Filatelele commented:
Очень глубокий WIP, так как нормальная реализация невозможна без #132 
Stay tuned.

Filatelele committed:
Merge branch 'master' of https://github.com/GammaStation/Gamma-Station into Holomaps
